sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1749,7 +1749,8 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
             m.copy(mergeCondition = resolvedMergeCondition,
               matchedActions = newMatchedActions,
               notMatchedActions = newNotMatchedActions,
-              notMatchedBySourceActions = newNotMatchedBySourceActions)
+              notMatchedBySourceActions = newNotMatchedBySourceActions,
+              originalSourceActions = newMatchedActions ++ newNotMatchedActions)
```

Comment:
note: because (not)matchedActions actions get changed by rule: ResolveRowLevelCommandAssignments, i need to preserve the original user actions/assignments so that MergeIntoTable.referencedSourceSchema and needsSchemaEvolution is idempotent

Reply:
shall we keep `referencedSourceSchema` directly?

Reply:
ok, discuss offline, with latest restrictions, no need for this now.